cmake/kokkos_arch.cmake

```diff
@@ -243,12 +243,12 @@ mark_as_advanced(Kokkos_IMPL_AMDGPU_LINK)
 #clear anything that might be in the cache
 global_set(KOKKOS_AMDGPU_OPTIONS)
 if(KOKKOS_ENABLE_HIP)
+  global_append(KOKKOS_AMDGPU_OPTIONS -xhip)
```

Comment:
I am not sure I am happy about this. The command line now has -x hip, -x c++, -x hip I assume with hipcc?

Reply:
I think we should for now just have CMP0119 OLD set. Everything else feels even more hacky than it should be.

Reply:
And lets complain to Kitware.

Reply:
> I am not sure I am happy about this. The command line now has -x hip, -x c++, -x hip I assume with hipcc?

It has -x c++ and -x hip. The commit introducing the policy change has a comment about the placement of the new flag so that it can be overridden.

Reply:
https://gitlab.kitware.com/cmake/cmake/-/commit/48aac247e922409b4dfe7189521661df4b0583d1#a103b2c88fe7a8431bd373f30d9a02764d41fc59

Reply:
hipcc itself calls amclang++ with -x hip, so I think it is at that level -x hip -x c++ -x hip or?

Reply:
but ok I guess so if its intended to be overriden and there aren't compilers which complain setting it multiple times in conflicting ways.